### PR TITLE
feat: Leverage new theme.elevations instead of theme.shadows

### PR DESCRIPTION
**Pull request by lukelooker**
- [x] 🖼 Image Snapshot coverage

### CODE REVIEW

File: packages/components/src/Overlay/OverlaySurface.tsx [hunk 1]
```diff
@@ -124,7 +124,7 @@ export const OverlaySurface = styled(OverlaySurfaceLayout)`
 export const OverlaySurfaceContentArea = styled.div`
   background: ${({ theme }) => theme.colors.background};
   border-radius: ${({ theme }) => theme.radii.medium};
-  box-shadow: ${({ theme }) => theme.shadows[3]};
+  box-shadow: ${({ theme }) => theme.elevations.plus3};
```
Comment (jhardy) [inline packages/components/src/Overlay/OverlaySurface.tsx:127]: does this get used for `Menu`? if possible, can we move it to `plus2` with the new shadows

File: packages/components/src/Menu/NestedMenuSurface.ts [hunk 2]
```diff
@@ -35,6 +35,6 @@ export const NestedMenuSurface = styled(OverlaySurface)`
     padding: 0;
   }
   ${OverlaySurfaceContentArea} {
-    box-shadow: ${({ theme }) => theme.shadows[1]};
+    box-shadow: ${({ theme }) => theme.elevations.plus1};
```
Comment (jhardy) [inline packages/components/src/Menu/NestedMenuSurface.ts:38]: I think nested menus should be at `plus2` as well to match the surface

File: packages/components/src/Dialog/DialogSurface.tsx [hunk 3]
```diff
@@ -121,7 +121,7 @@ export const DialogSurface = styled(SurfaceBase).attrs<DialogSurfaceProps>(
     width,
   })
 )<DialogSurfaceProps>`
-  box-shadow: ${({ theme }) => theme.shadows[5]};
+  box-shadow: ${({ theme }) => theme.elevations.plus5};
```
Comment (jhardy) [inline packages/components/src/Dialog/DialogSurface.tsx:124]: Any objections to going to `plus3` here?

**Review by jhardy — COMMENTED** [review]
Over all looks really good, just a few questions

File: packages/components/src/Overlay/OverlaySurface.tsx (hunk 1 above)
Comment (lukelooker) [reply packages/components/src/Overlay/OverlaySurface.tsx:127]: 👍  - changed, waiting on screenshot updates.

File: packages/components/src/Menu/NestedMenuSurface.ts (hunk 2 above)
Comment (lukelooker) [reply packages/components/src/Menu/NestedMenuSurface.ts:38]: 👍  - changed, waiting on screenshot updates.

File: packages/components/src/Dialog/DialogSurface.tsx (hunk 3 above)
Comment (lukelooker) [reply packages/components/src/Dialog/DialogSurface.tsx:124]: 👍  - changed, waiting on screenshot updates.

**Review by jhardy — APPROVED** [review]
LGTM

**Pull request merged**